dpath/segments.py

```diff
@@ -241,10 +245,13 @@ def match(segments: Path, glob: Glob):
                 s = str(s)
 
             try:
-                # Let's see if the glob matches. We will turn any kind of
-                # exception while attempting to match into a False for the
-                # match.
-                if not fnmatchcase(s, g):
+                # Let's see if the glob or the regular expression matches. We will turn any kind of
+                # exception while attempting to match into a False for the match.
+                if isinstance(g, Pattern):
```

Comment:
Is there any need to check `options.DPATH_ACCEPT_RE_REGEXP` at this point?

Reply:
No, there are 2 ways to receive a Pattern at this point:
- the user has explicitly introduced an re.Pattern using re.compile
- we have applied re.compile, but we do so only if options.DPATH_ACCEPT_RE_REGEXP is enabled.
So if we receive a Pattern at this point it is legal and we just deal with it.

Introduction of a Pattern in a path list by the user is backward compatible, since it was clearly forbidden before

However the variable should be renamed options.DPATH_ACCEPT_RE_REGEXP_IN_STRING, so that it is clear that it pertains only to regexp recognition.

.